### DISCUSSION TITLE
Atmoshere shader

### DISCUSSION THREAD

**Discussion by VecterraSoft** [Hoimar/Planet-Generator#41] [Q&A]
Hello, i was wondering what the for loop in the atmosphere shader does.
      
        int steps = 16;
	// TODO Some stuff can be optimized
	for (int i = 0; i < steps; ++i) {

It seems that when i lower the steps to only 2 i get the same result as 16 or 256. I see no real difference if any.
The only difference is the performance hit . I get a 15% fps bump when i set to 2.

Please advice.

**Comment by Hoimar** [comment]
It calculates attenuation of light along rays. It's odd that there's no difference visible.

It's here in the updated version: https://github.com/Zylann/godot_atmosphere_shader/blob/master/addons/zylann.atmosphere/shaders/include/atmosphere_funcs_v1.gdshaderinc

**Reply by SonomSoft** [reply]
Okay Thank you!, JUST TRIED THE NEW ATMOSPHERE sHADER AND IT WORKS SUPERB IN 4.3. I cost about a 17% fps hit, but ill tweek it to improve performance. 

![Screenshot from 2025-02-03 22-08-16](https://github.com/user-attachments/assets/6223e2d5-e35a-42e2-a161-535fe74e3103)

I must say though, that 17% hit is worth it looks amazing.
